unity/src/main/java/io/delta/unity/UCCatalogManagedCommitter.java

```diff
@@ -129,6 +132,89 @@ private CommitResponse writeImpl(
     return new CommitResponse(ParsedCatalogCommitData.forFileStatus(kernelStagedCommitFileStatus));
   }
 
+  /////////////////////////////
+  // Publish-related methods //
+  /////////////////////////////
+
+  @Override
+  public void publish(Engine engine, PublishMetadata publishMetadata)
+      throws PublishFailedException {
+    requireNonNull(engine, "engine is null");
+    requireNonNull(publishMetadata, "publishMetadata is null");
+
+    final List<ParsedCatalogCommitData> catalogCommits =
+        publishMetadata.getAscendingCatalogCommits();
+
+    if (catalogCommits.isEmpty()) {
+      return;
+    }
+
+    final String logPath = publishMetadata.getLogPath();
+    final long snapshotVersion = publishMetadata.getSnapshotVersion();
+
+    logger.info(
+        "[{}] Publishing {} catalog commits up to version {}",
+        ucTableId,
+        catalogCommits.size(),
+        snapshotVersion);
+
+    for (ParsedCatalogCommitData catalogCommit : catalogCommits) {
+      publishSingleCommit(engine, catalogCommit, logPath);
+    }
+
+    logger.info(
+        "[{}] Successfully published all catalog commits up to version {}",
+        snapshotVersion,
+        ucTableId);
+  }
+
+  private void publishSingleCommit(
+      Engine engine, ParsedCatalogCommitData catalogCommit, String logPath)
+      throws PublishFailedException {
+    final long commitVersion = catalogCommit.getVersion();
+
+    if (catalogCommit.isInline()) {
+      throw new UnsupportedOperationException(
+          "Publishing inline catalog commits is not yet supported");
+    }
+
+    final String sourcePath = catalogCommit.getFileStatus().getPath();
+    final String targetPath = FileNames.deltaFile(logPath, commitVersion);
+
+    try {
+      logger.info("[{}] Publishing catalog commit: {} -> {}", ucTableId, sourcePath, targetPath);
+
+      // Copy the staged commit file to the published delta file location.
+      // Use overwrite=false to ensure PUT-if-absent semantics.
+      engine
+          .getFileSystemClient()
+          .copyFileAtomically(sourcePath, targetPath, false /* overwrite */);
+
+      logger.info("[{}] Successfully published version {}", ucTableId, commitVersion);
+    } catch (java.nio.file.FileAlreadyExistsException e) {
+      // File already exists - this is okay, it means this version was already published
```

Comment:
so we want put-if-absent just so we can log it?

Reply:
Responded above 👍